### PR TITLE
gh-115664: Reorganize versionadded directives for -X options

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
Add the versionadded directives just under the description of the corresponding option.

It is an alternative to changes in #116298.

* Issue: gh-115664

----
📚 Documentation preview 📚: https://cpython-previews--116304.org.readthedocs.build/

### CODE REVIEW

File: Doc/using/cmdline.rst [hunk 1]
```diff
@@ -497,43 +497,80 @@ Miscellaneous options
...
      default.
+
+     .. versionadded:: 3.7
+
+     .. versionchanged:: 3.8
+        The :class:`io.IOBase` destructor logs ``close()`` exceptions.
+
+     .. versionchanged:: 3.9
+        Check the *encoding* and *errors* arguments for string encoding
+        and decoding operations.
```
Comment (serhiy-storchaka) [inline Doc/using/cmdline.rst]: I am not sure about these two. The Python Development Mode is now described in details in Doc/library/devmode.rst, where there are corresponding `versionchanged` directives. So perhaps there is no need to duplicate them here. Perhaps they are remnants from the time when it all was described here.

Comment (serhiy-storchaka) [inline Doc/using/cmdline.rst]:
```suggestion
```

File: Doc/using/cmdline.rst [hunk 2]
```diff
@@ -542,65 +579,40 @@ Miscellaneous options
...
-
-   .. versionchanged:: 3.8
-      Added the ``-X pycache_prefix`` option. The ``-X dev`` option now logs
-      ``close()`` exceptions in :class:`io.IOBase` destructor.
-
-   .. versionchanged:: 3.9
-      Using ``-X dev`` option, check *encoding* and *errors* arguments on
-      string encoding and decoding operations.
-
-      The ``-X showalloccount`` option has been removed.
```
Comment (hugovk) [inline Doc/using/cmdline.rst]: I think it's still useful to know `-X showalloccount` was removed in 3.9?

For example, if I'm migrating from 3.8 to 3.9, I might want to know this option has gone.

File: Doc/using/cmdline.rst [hunk 3]
```diff
@@ -542,65 +579,40 @@ Miscellaneous options
...
-
-   .. versionchanged:: 3.10
-      Added the ``-X warn_default_encoding`` option.
-      Removed the ``-X oldparser`` option.
```
Comment (hugovk) [inline Doc/using/cmdline.rst]: Similarly about `-X oldparser` being removed in 3.10.

**Review by hugovk — COMMENTED** [review]
Thanks, this is an improvement. A couple of questions about keeping info about removals.

File: Doc/using/cmdline.rst (hunk 2 above)
Comment (serhiy-storchaka) [reply Doc/using/cmdline.rst]: Done. Although it is slightly inconsistent. Either we document an entity (class, method) separately, with `versionadded`, and when remove it, nothing is left. Or we document it (function parameters) in the documentation of other entity, with `versionchanged` for the enclosing entity, and after removing them, we can add a new `versionchanged`. There is no strong division by types. Attributes can be documented in both ways, and sometimes methods and even classes are not documented as separate entities.

**Review by hugovk — APPROVED** [review]
Thanks!

**Comment by serhiy-storchaka** [comment]
Thank you for your review @hugovk.

**Pull request merged**